src/autoscaler/envelopeprocessor/envelope_processor.go

```diff
@@ -330,5 +309,12 @@ func processCustomMetrics(e *loggregator_v2.Envelope, currentTimestamp int64) ([
 			Timestamp:     e.Timestamp,
 		})
 	}
-	return metrics, nil
+	return metrics
+}
+
+func copyBaseAttributes(dst *models.AppInstanceMetric, src *models.AppInstanceMetric) {
+	dst.AppId = src.AppId
+	dst.InstanceIndex = src.InstanceIndex
+	dst.CollectedAt = src.CollectedAt
+	dst.Timestamp = src.Timestamp
```

Comment:
There is no need for a custom implementation to handle "attribute copying" in a generic way, as no other component requires or will require this functionality.

That's why I kept it simple and stupid!